extension_service/datastore/providers/firestore.py

```diff
@@ -0,0 +1,270 @@
+            },
+        )
+        return cls(firestore.AsyncClient(project=config.projectId))
+
+    async def initialize_data(
+        self,
+        airports: list[models.Airport],
+        amenities: list[models.Amenity],
+        flights: list[models.Flight],
+    ) -> None:
+        async def delete_collection(coll_ref, batch_size=400):
+            docs = coll_ref.limit(batch_size).stream()
+            deleted = 0
+
+            async for doc in docs:
+                await doc.reference.delete()
+                deleted = deleted + 1
+
+            if deleted >= batch_size:
+                return await delete_collection(coll_ref, batch_size)
+
+        # Check if the collections already exist; if so, delete collections
+        airports_ref = self.__client.collection("airports")
+        airports_exist = await airports_ref.limit(1).get()
+        if airports_exist:
+            await delete_collection(airports_ref)
+
+        amenities_ref = self.__client.collection("amenities")
+        amenities_exist = await amenities_ref.limit(1).get()
+        if amenities_exist:
+            await delete_collection(amenities_ref)
+
+        flights_ref = self.__client.collection("flights")
+        flights_exist = await flights_ref.limit(1).get()
+        if flights_exist:
+            await delete_collection(flights_ref)
+
+        # initialize collections
+        for airport in airports:
+            await self.__client.collection("airports").document(str(airport.id)).set(
+                {
+                    "iata": airport.iata,
+                    "name": airport.name,
+                    "city": airport.city,
+                    "country": airport.country,
+                }
+            )
+        for amenity in amenities:
+            await self.__client.collection("amenities").document(str(amenity.id)).set(
+                {
+                    "name": amenity.name,
+                    "description": amenity.description,
+                    "location": amenity.location,
+                    "terminal": amenity.terminal,
+                    "category": amenity.category,
+                    "hour": amenity.hour,
+                    "content": amenity.content,
+                    "embedding": amenity.embedding,
+                }
+            )
+        for flight in flights:
+            await self.__client.collection("flights").document(str(flight.id)).set(
+                {
+                    "airline": flight.airline,
+                    "flight_number": flight.flight_number,
+                    "departure_airport": flight.departure_airport,
+                    "arrival_airport": flight.arrival_airport,
+                    "departure_time": flight.departure_time,
+                    "arrival_time": flight.arrival_time,
+                    "departure_gate": flight.departure_gate,
+                    "arrival_gate": flight.arrival_gate,
+                }
+            )
```

Comment:
This seems like another good target for asyncronization

Reply:
Updated with async.